src/server/components/dashboard/listsItems/controllers.ts

```diff
@@ -209,62 +170,37 @@ export async function listItemPostController(
   });
 }
 
-export async function listItemPinController(
-  req: Request,
-  res: Response
-): Promise<void> {
+export async function listItemPinController(req: Request, res: Response): Promise<void> {
   const { listId, listItemId } = req.params;
   const { action } = req.body;
   const userId = req?.user?.userData?.id;
   const isPinned = action === "pin";
 
   // eslint-disable-next-line @typescript-eslint/consistent-type-assertions
-  const listItem: ListItemGetObject = (await findListItemById(
-    listItemId
-  )) as ListItemGetObject;
+  const listItem: ListItemGetObject = (await findListItemById(listItemId)) as ListItemGetObject;
 
   if (userId === undefined) {
-    req.flash(
-      "errorMsg",
-      "Unable to perform action - user could not be identified"
-    );
-    return res.redirect(
-      dashboardRoutes.listsItem
-        .replace(":listId", listId)
-        .replace(":listItemId", listItemId)
-    );
+    req.flash("errorMsg", "Unable to perform action - user could not be identified");
+    return res.redirect(dashboardRoutes.listsItem.replace(":listId", listId).replace(":listItemId", listItemId));
   }
 
   try {
     await handlePinListItem(Number(listItemId), userId, isPinned);
 
     req.flash(
       "successBannerTitle",
-      `${listItem.jsonData.organisationName} has been ${
-        isPinned ? "pinned" : "unpinned"
-      }`
+      `${listItem.jsonData.organisationName} has been ${isPinned ? "pinned" : "unpinned"}`
     );
     req.flash("successBannerHeading", `${isPinned ? "Pinned" : "Unpinned"}`);
     req.flash("successBannerColour", "blue");
     res.redirect(dashboardRoutes.listsItems.replace(":listId", listId));
   } catch (error: any) {
-    req.flash(
-      "errorMsg",
-      `${listItem.jsonData.organisationName} could not be updated. ${error.message}`
-    );
-    return res.redirect(
-      dashboardRoutes.listsItem
-        .replace(":listId", listId)
-        .replace(":listItemId", listItemId)
-    );
+    req.flash("errorMsg", `${listItem.jsonData.organisationName} could not be updated. ${error.message}`);
+    return res.redirect(dashboardRoutes.listsItem.replace(":listId", listId).replace(":listItemId", listItemId));
   }
 }
 
-export async function handlePinListItem(
-  id: number,
-  userId: User["id"],
-  isPinned: boolean
-): Promise<ListItem> {
+export async function handlePinListItem(id: number, userId: User["id"], isPinned: boolean): Promise<ListItem> {
   if (userId === undefined) {
```

Comment:
Fair point, for this one it would be easier for me to put in the listItemPinController() via a middleware like I've done to the other controllers. Unless you think the handlePinListItem() is a special case?